src/volume_mgr.rs

```diff
@@ -775,14 +775,36 @@ where
 
     /// Close a file with the given raw file handle.
     pub fn close_file(&mut self, file: RawFile) -> Result<(), Error<D::Error>> {
-        let file_idx = self.flush_file_get_index(file)?;
+        let flush_result = self.flush_file(file);
+        let file_idx = self.get_file_by_id(file)?;
```

Comment:
It’s unfortunate to have to find the file index twice.

Reply:
Yeah I thought about that. But, I also think the previous flush_file_get_index was a bit of an awkward dual-purpose function. This implementation is ever so slightly less performant at closing a file, but the functions and their purpose are clearer.

Reply:
This is persuasive